java/spark-1.5.2/backend.scala

```diff
@@ -276,6 +283,13 @@ class Backend {
                     logger.log("is terminating")
                     System.exit(0)
                   }
+
+                  // workers should always terminate but without exceptions
+                  if (isWorker) {
+                    logger.log("is terminating worker")
```

Comment:
Maybe a more informative log message: `sparklyr worker backend terminating` or something like that?